Content/Items/Breacher/Weapons.BreachCannon.cs

```diff
@@ -0,0 +1,160 @@
+	{
+		public override string Texture => AssetDirectory.BreacherItem + Name;
+
+		public override void SetStaticDefaults()
+		{
+			DisplayName.SetDefault("Breach Cannon");
+			Tooltip.SetDefault("Summons a sentry that shoots a laser towards the cursor \nThese lasers can combine");
+		}
+
+		public override void SetDefaults()
+		{
+			Item.CloneDefaults(ItemID.QueenSpiderStaff);
+			Item.damage = 12;
+			Item.mana = 12;
+			Item.width = 40;
+			Item.height = 40;
+			Item.rare = ItemRarityID.Orange;
+			Item.value = Item.sellPrice(0, 1, 0, 0);
+			Item.knockBack = 2.5f;
+			Item.UseSound = SoundID.Item25;
+			Item.useStyle = ItemUseStyleID.HoldUp;
+			Item.shoot = ModContent.ProjectileType<BreachCannonSentry>();
+			Item.shootSpeed = 0f;
+		}
+
+		public override bool Shoot(Player player, EntitySource_ItemUse_WithAmmo source, Vector2 position, Vector2 velocity, int type, int damage, float knockback)
+		{
+			position = Main.MouseWorld;
+			Tile startTile = Main.tile[(int)(position.X / 16), (int)(position.Y / 16)];
+
+			if (startTile.HasTile && Main.tileSolid[startTile.TileType])
+				return false;
+
+			//0 = right
+			//1 = bottom
+			//2 = left
+			//3 = top
+
+			int minDistance = 99;
+			int minDirection = 0;
+
+			Vector2 originTile = Vector2.Zero;
+
+			for (int k = 0; k < 4; k++)
+			{
+				Vector2 testPosition = position / 16;
+				testPosition.X = (int)testPosition.X;
+				testPosition.Y = (int)testPosition.Y;
+				Vector2 direction = Vector2.UnitX.RotatedBy(k * 1.57f);
+				int testDistance = 0;
+				while (testDistance < 40)
+				{
+					testPosition += direction;
+					testDistance++;
+					if (testDistance > minDistance)
+						break;
+
+					int i = (int)testPosition.X;
+					int j = (int)testPosition.Y;
+					Tile testTile = Main.tile[i, j];
+					if (testTile.HasTile && Main.tileSolid[testTile.TileType])
+					{
+						minDistance = testDistance;
+						minDirection = k;
+						originTile = testPosition;
+						break;
+					}
+				}
+			}
+
+			if (minDistance == 99)
+				return false;
+
+			originTile = MoveRightAndDown(originTile, minDirection);
+			if (originTile == Vector2.Zero)
+				return false;
+
+			Projectile proj = Projectile.NewProjectileDirect(source, (originTile - Vector2.UnitX.RotatedBy(minDirection * 1.57f)) * 16, velocity, type, damage, knockback, player.whoAmI, minDirection);
+			var mp = proj.ModProjectile as BreachCannonSentry;
+
+			//Main.NewText("Spawned at (" + ((int)originTile.X).ToString() + "," + ((int)originTile.Y).ToString() + ")");
+			mp.tileOrigin = originTile;
+			proj.originalDamage = Item.damage;
+			proj.rotation = (minDirection * 1.57f) + 3.14f;
+			player.UpdateMaxTurrets();
+
+			return false;
+		}
+
+		public override void AddRecipes()
+		{
+			Recipe recipe = CreateRecipe();
+			recipe.AddIngredient(ModContent.ItemType<Items.SpaceEvent.Astroscrap>(), 12);
+			recipe.AddTile(TileID.Anvils);
+			recipe.Register();
+		}
+
+		private Vector2 MoveRightAndDown(Vector2 originTile, int minDirection)
```

Comment:
Is there a reason we explicitly favor the right hand side? Should we not favor the closest position instead?

Reply:
I genuinely don't know how you'd do the closest position. Right was chosen arbitrarily, but considering how niche this is, I'm not sure it's super important to fix. However, if you have an idea for how to adapt this method to do closest, let me know.

Reply:
Would simply alternating between left and right not work? check x+1, x-1, x+2, x-2, etc.?